internal/cli/cmd/apply/apply.go

```diff
@@ -19,8 +19,54 @@ import (
 	kargosvcapi "github.com/akuity/kargo/pkg/api/service/v1alpha1"
 )
 
+type applyOptions struct {
+	*option.Option
+
+	Filenames []string
+}
+
+// newApplyOptions creates a new options struct.
+func newApplyOptions(opt *option.Option) *applyOptions {
+	return &applyOptions{Option: opt}
+}
+
+// AddFlags adds the flags for the apply options to the provided command.
+func (o *applyOptions) AddFlags(cmd *cobra.Command) {
+	o.PrintFlags.AddFlags(cmd)
+
+	// TODO: Factor out server flags to a higher level (root?) as they are
+	//   common to almost all commands.
```

Comment:
Ya... it's tricky. _Almost but not all_ commands use them.

Personally, I'd rather repeat ourselves sometimes that have inapplicable flags show up on commands. i.e. If we had to choose, UX wins over DRY, imho.

Reply:
A problem I have observed though is that we sometimes ourselves forget to actually wire them in (see e.g. https://github.com/akuity/kargo/blob/5912e5a9dfd2912fdd6a46e1b35655b6225c4454/internal/cli/cmd/approve/approve.go in current `main`, which AFAIK should have the flags set).

By registering them once at root and loading them into some config which is passed to downward commands, this could be avoided.

Reply:
Is there a way to do it at root and _unregister_ them in spots?

Reply:
I think you can theoretically use `MarkHidden` to hide them from the help menu for specific commands.